lightning/src/ln/channel.rs

```diff
@@ -2905,14 +2913,15 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 		}
 		assert_eq!(self.channel_state & ChannelState::ShutdownComplete as u32, 0);
 
-		// BOLT 2 says we must only send a scriptpubkey of certain standard forms, which are up to
-		// 34 bytes in length, so don't let the remote peer feed us some super fee-heavy script.
-		if self.is_outbound() && msg.scriptpubkey.len() > 34 {
+		// BOLT 2 says we must only send a scriptpubkey of certain standard forms,
+		// which for a a BIP-141-compliant witness program is at max 42 bytes in length.
+		// So don't let the remote peer feed us some super fee-heavy script.
+		if self.is_outbound() && msg.scriptpubkey.len() > 42 {
```

Comment:
Hmm, I dont see why this check had an `is_outbound()` on it - it didn't matter because the script had to be a more concrete (and small) type in the next check, but now it does. This check should probably also move into `unsupported_witness_shutdown_script` IMO.